### ISSUE TITLE
Pixi offline upgrading and everything else

### ISSUE THREAD

**Issue by toiletsandpaper** [prefix-dev/pixi#1765]
### Problem description

Hi! Pixi is a great tool, I'm enjoing using it on my personal laptops and other.

But using pixi for serious commercial projects with cybersec limitations is a huge pain. For example, I have disconnected all my GPU clusters from internet access and setted up mirrors and cachers: Sonatype Nexus, Harbor, etc.; so I can work in a controlled environment. And other tools, like docker, podman, pdm, poetry, pip, conda/mamba is ready and can be upgraded with `apt`/`dnf`/`pipx` - pixi is not (self-upgrade requires internet access and https://github.com/prefix-dev/pixi/issues/1764)

I see only two ways to fix this:
1. Write and open-source your own Nexus-like tool for mirror and cache packages (maybe like `panamax` for Rust crates)
2. Make pixi offline-friendly and also distribute it with apt/dnf/pipx/etc. 

Anyways, thank you for your work! Great tool!

**Comment by pavelzw** [comment]
getting pixi into apt and dnf would be great 👍🏻 not sure how the rules for these packaging ecosystems are, though. you can definitely maintain your own packages but it would be better if we managed to get them upstream into the main repos

**Comment by ruben-arts** [comment]
You can setup a selfhosted conda package mirror with [Quetz](https://github.com/mamba-org/quetz)

I'd be open to get pixi on `pypi` but currently don't have the capacity. `apt` or `dnf` would be even harder for me.

That said contributions are always welcome! So if anyone has the time and knowledge please help us get `pixi` distributed though other ecosystems.

**Comment by toiletsandpaper** [comment]
> You can setup a selfhosted conda package mirror with Quetz

I have a problem with this. You can check my other opened issue, that I've linked in topic message, I've explained everything there.

**Comment by ruben-arts** [comment]
> I have a problem with this. You can check my other opened issue, that I've linked in topic message, I've explained everything there.

This issue in #1764  is unrelated to the self-hosted conda package mirror. Let me explain that issue in #1764.

**Comment by toiletsandpaper** [comment]
> @toiletsandpaper would you also require like an `--offline` mode, where it does not touch the internet during pixi operations as well?

What kind of operation requires internet connection? I don't really think that tools like pixi really should require internet at all. Maybe for `self-update` or initial installation, but nothing else.

I saw pixi trying to dowload pypi-mappings from github. If you want to keep it up to date even if user don't want to upgrade pixi - yeah, `--offline` mode that forces pixi to disable mapping updating would be great. Maybe even as a flag in `.config` folder?

**Comment by tdejager** [comment]
Well, a lot of people like the option of having up to date packages and index data by default 😄 This is how most package managers operate. However, thats why `uv` and other tools offer a `--offline` flag.

**Comment by toiletsandpaper** [comment]
> Well, a lot of people like the option of having up to date packages and index data by default 😄 This is how most package managers operate. However, thats why `uv` and other tools offer a `--offline` flag.

Yeah, I get it, but I have pypi and conda mirrors and `--offline` will ignore my mirrors too, right? If so, then `--offline` is not that relevant in my case, idk about others :)

**Comment by ruben-arts** [comment]
If you only have conda dependencies you can use [`pixi-pack`](https://github.com/Quantco/pixi-pack) which is the equivalent of conda-pack.

**Comment by toiletsandpaper** [comment]
I do really hope that someone with good rust expirience can make use of https://github.com/PyO3/maturin that was built to publish rust tools on PyPI. The closes example is https://github.com/astral-sh/uv as it can be downloaded via pip (`pipx install uv`)
